From bbd438309258243608dd08e90725646d17572810 Mon Sep 17 00:00:00 2001
From: Nickey Yang <nickey.yang@rock-chips.com>
Date: Mon, 29 Jun 2020 20:47:20 +0800
Subject: [PATCH] ARM: dts: rockchip: rv11xx-evb-v10: add GT917S tp support

Gt917s is a panel adapter board with ili9881d matching touch IC.

At the same time, init delay Ms 80 instead of 10, is to solve the
problem of high probability of not displaying due to incomplete
reset when the hardware of the panel adapter board is changed.

Signed-off-by: Nickey Yang <nickey.yang@rock-chips.com>
Change-Id: I17eff7fdd140a7d6f82e8991ad2bdd7765d43a73
---
 arch/arm/boot/dts/rv11xx-evb-v10.dtsi | 24 +++++++++++++++++++++++-
 1 file changed, 23 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/rv11xx-evb-v10.dtsi b/arch/arm/boot/dts/rv11xx-evb-v10.dtsi
index 11585b1a88de..2882c7b50222 100644
--- a/arch/arm/boot/dts/rv11xx-evb-v10.dtsi
+++ b/arch/arm/boot/dts/rv11xx-evb-v10.dtsi
@@ -236,7 +236,7 @@
 		power-supply = <&vcc18_lcd_n>;
 		prepare-delay-ms = <5>;
 		reset-delay-ms = <1>;
-		init-delay-ms = <10>;
+		init-delay-ms = <80>;
 		disable-delay-ms = <10>;
 		unprepare-delay-ms = <5>;
 
@@ -929,6 +929,21 @@
 	};
 };
 
+&i2c5 {
+	status = "okay";
+	clock-frequency = <400000>;
+
+	gt1x: gt1x@5d {
+		compatible = "goodix,gt1x";
+		reg = <0x5d>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&tp_int_pin>;
+		power-supply = <&vcc18_lcd_n>;
+		goodix,rst-gpio = <&gpio2 RK_PB0 GPIO_ACTIVE_HIGH>;
+		goodix,irq-gpio = <&gpio2 RK_PB2 GPIO_ACTIVE_LOW>;
+	};
+};
+
 &i2s0_8ch {
 	status = "okay";
 	#sound-dai-cells = <0>;
@@ -1076,6 +1091,13 @@
 			rockchip,pins = <0 RK_PB0 0 &pcfg_pull_up>;
 		};
 	};
+
+	gt1x {
+		/omit-if-no-ref/
+		tp_int_pin: tp-int-pin {
+			rockchip,pins = <2 RK_PB2 RK_FUNC_GPIO &pcfg_pull_up>;
+		};
+	};
 };
 
 &pmu_io_domains {
-- 
2.35.3

